server_vllm.py

```diff
@@ -334,9 +334,27 @@ async def completion_stream_generator(
         tool_choice, functions
     ) -> AsyncGenerator[str, None]:
         generator = wrap_vllm_generator(tool_choice=tool_choice)
+        tool_call_count = 0
         async for response in generate_openai_format_from_stream_async(
             generator, prompt_template, tool_choice
         ):
+            # Convert tool_calls to function_call if request.functions is provided
+            if (
+                functions
+                and len(functions) > 0
+                and "tool_calls" in response["delta"]
+                and response["delta"]["tool_calls"]
+                and len(response["delta"]["tool_calls"]) > 0
+            ):
+                tool_name = response["delta"]["tool_calls"][0]["function"]["name"]
+                tool_args = response["delta"]["tool_calls"][0]["function"]["arguments"]
+                if tool_name and len(tool_name) > 0 and tool_args == "":
```

Comment:
we can remove this : tool_args=="" in this condition

Reply:
It's just additional check though